solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -236,8 +240,11 @@ protected ZkCredentialsProvider createZkCredentialsToAddAutomatically() {
     if (!StringUtils.isEmpty(zkCredentialsProviderClassName)) {
       try {
         log.info("Using ZkCredentialsProvider: {}", zkCredentialsProviderClassName);
-        return (ZkCredentialsProvider)
-            Class.forName(zkCredentialsProviderClassName).getConstructor().newInstance();
+        ZkCredentialsProvider zkCredentialsProvider =
+            (ZkCredentialsProvider)
+                Class.forName(zkCredentialsProviderClassName).getConstructor().newInstance();
```

Comment:
We can't use Class's classloader here. At the very least we need to use `Thread.currentThread().getContextClassLoader().loadClass(zkCredentialsProviderClassName)`.

Same for the zkAclProvider below

Reply:
This an the existing code, not part of this PR, are you suggesting to change that?